massa-execution-worker/src/execution.rs

```diff
@@ -1470,16 +1470,25 @@ impl ExecutionState {
                         if prev.timestamp.eq(&MassaTime::from(0)) {
                             // first range with timestamp = 0
                             prev.start_slot = Slot::min();
-                        } else {
-                            prev.start_slot = get_slot_at_timestamp(config, prev.timestamp)?;
+                        } else if let Some(slot) = get_slot_at_timestamp(config, prev.timestamp)? {
+                            prev.start_slot = slot;
                         }
 
                         // retrieve the end_slot
-                        let next_range_slot = get_slot_at_timestamp(config, next.timestamp)?;
-                        prev.end_slot = next_range_slot.get_prev_slot(config.thread_count)?;
+                        if let Some(next_range_slot) =
+                            get_slot_at_timestamp(config, next.timestamp)?
+                        {
+                            prev.end_slot = next_range_slot.get_prev_slot(config.thread_count)?;
+                        }
 
                         Ok(prev)
                     })
+                    // we don't need range with StartSlot(0,0) && EndSlot(0,0)
```

Comment:
We need the range with the start timestamp at genesis otherwise from genesis to X there is no vesting.

Reply:
No problem.